include/nfc/tnep/tag.rst

```diff
@@ -0,0 +1,141 @@
+
+To use the NDEF message buffers, you must configure the following elements:
+
+* Provide a pointer to two TX buffers - one for emulating the NDEF message and one for encoding the new NDEF message.
+* Set a maximum buffer size in bytes, which is valid for both buffers.
+
+The buffers must be linked with the tag layer in the application.
+
+When the polling device writes a new NDEF message to the tag buffer, the application must inform the tag library by calling :cpp:func:`nfc_tnep_tag_rx_msg_indicate`.
+
+Defining TNEP services
+======================
+
+TNEP services are capabilities and features provided to the adjacent upper layer.
+They must be defined for the tag library to create the initial NDEF message.
+At least one TNEP service must be defined for the message to be created.
+
+To create a TNEP service, use the :c:macro:`NFC_TNEP_TAG_SERVICE_DEF` macro.
+
+After defining the services, create a table that lists all of them.
+You can use :c:macro:`NFC_TNEP_TAG_SERVICE` to get the names of the created services.
+
+The service table is given to the :cpp:func:`nfc_tnep_tag_initial_msg_create` function and the Service Parameter record of each service is inserted into the initial TNEP message.
+A Service Parameter record contains parameters for communicating with the service.
+
+See the following code for an example of service definition and TNEP initialization:
+
+.. code-block:: c
+
+	#define NFC_TNEP_TAG_SERVICE_DEF(service_name, uri, uri_length, mode, t_wait, n_wait, mac_msg_size
+								 select_cb, deselect_cb message_cb, timeout_cb, error_cb)
+
+	NFC_NDEF_MSG_DEF(ndef_msg, 16);
+
+	static struct nfc_tnep_tag_service services[] = {
+		NFC_TNEP_TAG_SERVICE(service_name),
+	};
+
+	nfc_tnep_tx_msg_buffer_register(buffer, swap_buffer, buffer_length);
+
+	nfc_tnep_init(event, event_cnt, &NFC_NDEF_MSG(ndef_msg),
+		      nfc_txt_rw_payload_set);
+
+	nfc_tXt_setup();
+
+	nfc_tnep_tag_initial_msg_create(services, ARRAY_SIZE(services), NULL, 0);
+
+	nfc_tXt_emulation_start();
+
+Configuration
+*************
+After initialization, the tag library is in the Service Ready state and the NDEF message buffer contains Service Parameter records provided with the initial TNEP message.
+
+The initial TNEP message can also include optional NDEF records that can interact with the polling device that does not support the protocol.
+
+Make sure to call the :c:func:`nfc_tnep_tag_process` function in an infinite loop in the application.
+
+Receiving new messages
+**********************
+
+When a new NDEF message appears in the buffer and it contains a Service Select record, the application can select this service.
+To do this, the application should inform the tag library of the service by calling :cpp:func:`nfc_tnep_tag_rx_msg_indicate`.
+The tag library will then change its state to Service Selected.
+From that point, the service decryption message will not longer be provided.
```

Comment:
It should point to TNEP Initial NDEF Message. It might be worth to change this description to make it easier to understand. But If service will be deselected then Tag return to Initial Message

Reply:
I don't understand - can you make a suggestion?